build_tools/github_actions/write_torch_versions.py

```diff
@@ -95,7 +95,11 @@ def main(argv: list[str]):
     p.add_argument(
         "--dist-dir",
         type=Path,
-        default=Path(os.getenv("PACKAGE_DIST_DIR")),
+        default=Path(
+            os.getenv("PACKAGE_DIST_DIR")
+            if not os.getenv("PACKAGE_DIST_DIR") == None
+            else "/no/dist-dir/or/env/PACKAGE_DIST_DIR/given"
+        ),
```

Comment:
I'm a bit confused by the changes here.

* For one, we could pull `os.getenv("PACKAGE_DIST_DIR")` into a local variable so it isn't repeated (general best practice for performance: avoid repeating syscalls or any other look up that is more expensive than some arithmetic)
* Then, the `else` value here of `/no/dist-dir/or/env/PACKAGE_DIST_DIR/given` would produce errors at runtime, right? In that case shouldn't we log a clear error message immediately and say that the argument is required and it was not specified? I think we can get that behavior with the existing code but then checking `if not args.dist_dir`, which we sort of already do below where we raise `FileNotFoundError`. Maybe that should first check `args.dist_dir` being `None` before checking `.exists()` on it?

Reply:
this was the minimal change version. if neither `PACKAGE_DIST_DIR` or `--dist-dir` exist, the `args.dist_dir.exists()` would file with 
```
Traceback (most recent call last):
  File "/home/TheRock/build_tools/github_actions/write_torch_versions.py", line 114, in <module>
    main(sys.argv[1:])
  File "/home/TheRock/build_tools/github_actions/write_torch_versions.py", line 107, in main
    raise FileNotFoundError(f"Dist dir '{args.dist_dir}' does not exist")
FileNotFoundError: Dist dir '/no/dist-dir/or/env/PACKAGE_DIST_DIR/given' does not exist
```
but i can make it fail nicer